bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/rest/DeviceProperty.java

```diff
@@ -0,0 +1,174 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public abstract sealed class DeviceProperty<T> implements Comparable<DeviceProperty<?>> {
+
+    private final @NotNull String name;
+    private final Boolean readOnly;
+    @Nullable
+    private final String direction;
+    @Nullable
+    private final String dataUpdatedAt;
+    @Nullable
+    private final String productName;
+    @Nullable
+    private final String displayName;
+    private T value;
+    private final Map<String, @Nullable Object> properties;
+
+    protected DeviceProperty(String name, @Nullable Boolean readOnly, @Nullable String direction,
+            @Nullable String dataUpdatedAt, @Nullable String productName, @Nullable String displayName,
+            @Nullable T value, @Nullable Map<String, @Nullable Object> properties) {
+        this.name = requireNonNull(name, "name cannot be null!");
+        this.readOnly = readOnly != null ? readOnly : true;
+        this.direction = direction;
+        this.dataUpdatedAt = dataUpdatedAt;
+        this.productName = productName;
+        this.displayName = displayName;
+        this.value = requireNonNull(value, "value");
+        this.properties = properties != null ? properties : Map.of();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Boolean getReadOnly() {
+        return readOnly;
+    }
+
+    @Nullable
+    public String getDirection() {
+        return direction;
+    }
+
+    @Nullable
+    public String getDataUpdatedAt() {
+        return dataUpdatedAt;
+    }
+
+    @Nullable
+    public String getProductName() {
+        return productName;
+    }
+
+    @NotNull
+    public String getDisplayName() {
+        var dn = displayName;
+        return dn != null ? dn : name;
+    }
+
+    public T getValue() {
+        return value;
+    }
+
+    public void setValue(T value) {
+        this.value = value;
+    }
+
+    public Map<String, @Nullable Object> getProperties() {
+        return properties;
+    }
+
+    @Override
+    public boolean equals(@Nullable Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        DeviceProperty<?> that = (DeviceProperty<?>) o;
+
+        return name.equals(that.name);
+    }
+
+    @Override
+    public int hashCode() {
+        return name.hashCode();
+    }
+
+    @Override
+    public int compareTo(DeviceProperty<?> o) {
+        return name.compareTo(o.name);
+    }
+
+    @Override
+    public String toString() {
+        return "DeviceProperty{" + "name='" + name + '\'' + ", readOnly=" + readOnly + ", direction='" + direction
+                + '\'' + ", value=" + value + '}';
+    }
+
+    /**
+     * @author Martin Grześlowski - Initial contribution
```

Comment:
I think putting an author tag in function doc is uncommon.... Up to you....

Reply:
It's inner class